kt/godot-runtime/src/main/kotlin/godot/core/Variant.kt

```diff
@@ -0,0 +1,284 @@
+            }
+    ),
+    QUAT(
+            { buffer: ByteBuffer ->
+                val x = buffer.float.toRealT()
+                val y = buffer.float.toRealT()
+                val z = buffer.float.toRealT()
+                val w = buffer.float.toRealT()
+
+                Quat(x, y, z, w)
+            },
+            { buffer: ByteBuffer, any: Any ->
+                any as Quat
+                buffer.putFloat(any.x.toFloat())
+                buffer.putFloat(any.y.toFloat())
+                buffer.putFloat(any.z.toFloat())
+                buffer.putFloat(any.w.toFloat())
+            }
+    ), // 10
+    AABB(
+            { buffer: ByteBuffer ->
+                val position = buffer.vector3
+                val size = buffer.vector3
+                AABB(position, size)
+            },
+            { buffer: ByteBuffer, any: Any ->
+                any as godot.core.AABB
+                buffer.vector3 = any._position
+                buffer.vector3 = any._size
+            }
+    ),
+    BASIS(
+            { buffer: ByteBuffer -> buffer.basis },
+            { buffer: ByteBuffer, any: Any -> buffer.basis = any as Basis }
+    ),
+    TRANSFORM(
+            { buffer: ByteBuffer ->
+                val basis = buffer.basis
+                val origin = buffer.vector3
+                Transform(basis, origin)
+            },
+            { buffer: ByteBuffer, any: Any ->
+                any as Transform
+                buffer.basis = any._basis
+                buffer.vector3 = any._origin
+            }
+    ),
+
+    // misc types
+    COLOR(
+            { buffer: ByteBuffer -> TODO() },
+            { buffer: ByteBuffer, any: Any -> TODO() }
+    ),
+    NODE_PATH(
+            { buffer: ByteBuffer -> TODO() },
+            { buffer: ByteBuffer, any: Any -> TODO() }
+    ), // 15
+    _RID(
+            { buffer: ByteBuffer -> TODO() },
+            { buffer: ByteBuffer, any: Any -> TODO() }
+    ),
+    OBJECT(
+            { buffer: ByteBuffer ->
+                val ptr = buffer.long
+                val constructorIndex = buffer.int
+                val isRef = buffer.bool
+                if (isRef) {
+                    GarbageCollector.getRefInstance(ptr)
+                } else {
+                    GarbageCollector.getObjectInstance(ptr)
+                } ?: KtObject.instantiateWith(
+                        ptr,
+                        buffer.long,
+                        isRef,
+                        TypeManager.engineTypesConstructors[constructorIndex]
+                )
+            },
+            { buffer: ByteBuffer, any: Any ->
+                any as KtObject
+                buffer.putLong(any.rawPtr)
+                buffer.bool = any.isRef
+            }
+    ),
+    DICTIONARY(
+            { buffer: ByteBuffer -> TODO() },
+            { buffer: ByteBuffer, any: Any -> TODO() }
+    ),
+    ARRAY(
+            { buffer: ByteBuffer -> VariantArray() },
```

Comment:
This impl isn't correct isn't it? It's just a placeholder right? If so maybe add a Todo (that does not crash with not yet implemented) so we don't forget to change it.